### ISSUE TITLE
Complete HCA reference refactor work

### ISSUE THREAD

**Issue by kbergin** [HumanCellAtlas/secondary-analysis#788]
Long story short: In Skylab accessory_workflows, there are scripts and WDLs for generating reference data. These need to be extended to support mouse reference data for SS2

The build indices workflow creates references for SS2 and Optimus for both human and mouse.  After the bulk of the work during the pipeline surge, there were a few steps remaining for the SS2 reference. The major changes that are added to complete the work is  

a) added a task for creating refflat file for Picard and the corresponding dockerfile; 

b) added a task for creating the interval list file for Picard; and 

c) corrected some naming problem in the initial pr [https://github.com/HumanCellAtlas/skylab/pull/204](https://github.com/HumanCellAtlas/skylab/pull/204)

As of now, all steps are complete.

**Why?** Complete work from the surge on SS2

**Where to start?** Sync with Phil or Ambrose. 

**ACs:**

* new WDL for generating a mouse reference is completed and merged in
* Update documentation for mouse for SS2

┆Issue is synchronized with this [Jira Dev Task](https://broadinstitute.atlassian.net/browse/GL-339)

**Comment by kbergin** [comment]
Moving testing AC to a separate ticket.

**Comment by kbergin** [comment]
It’d be great if we could close this out this week\!

**Comment by kbergin** [comment]
➤ Kishori Konwar commented:

Waiting for Jessica's approval. Hopefully we can close it tomorrow.

On Wed, Sep 11, 2019, 5:59 PM Kylee Degatano (JIRA Cloud) <

**Comment by kbergin** [comment]
➤ Nick Barkas commented:

Reviewed, PR needs a bit of work as it contains alot of things that are not directly relevant to this ticket. [~accountid:557058:f8ae8b22-a787-4422-a932-2c555c311cc4] volunteered to look into this (specifically splitting out the relevant bits). I will review again once this is done. I am moving this back to in progress.

**Comment by kbergin** [comment]
[~accountid:557058:f8ae8b22-a787-4422-a932-2c555c311cc4] there’s another ticket for ss2 mouse test already It’s in to do col

**Comment by kbergin** [comment]
➤ Nick Barkas commented:

I have gone ahead and deleted all the duplicated code after checking it does the same thing. I have also added the reference fasta and gtf as outputs to the pipeline, as these are frequently required as inputs to any pipeline that is using the reference.

I don’t think the folder needs renaming.

I think all the remaining tasks should be separate tickets, along with a few new ones. Specifically, the tickets we need are:

  * Update the mouse testing workflow to use the new style of tests
  * Add control for which references should be generated at the pipeline level
  * Generate indices for our pipelines using this new WDL and verify that the output is correct, while documenting the process. This should probably be done as part when the location of the production references has been identified.

[~accountid:557058:7f318d05-6e5c-4289-a9e0-ef0441bd7e57] do you know which of these are already tickets and which ones I should make?

**Comment by kbergin** [comment]
➤ Nick Barkas commented:

I went ahead and approved the PR

**Comment by kbergin** [comment]
Thanks Nick; 

  * Already a ticket I believe. Check to do column. 
  * Not sure what you mean here
  * there’s not a ticket for generating the references using the new wdl, nor documenting. There is for figuring out where to put them.

**Comment by kbergin** [comment]
➤ Nick Barkas commented:

Regarding 2, its just adding some switches to be able to generate only one of the references if required.

I made new tickets and moving this to QA

**Comment by kbergin** [comment]
➤ Nick Barkas commented:

QA:

  * Confirm the original ACs of the ticket are fullfiled (esp wrt documentation needs)
  * Check new tickets in backlog
